ikabot/function/distributeResources.py

```diff
@@ -158,99 +171,104 @@ def distribute_evenly(s, resource):
 
 	return routes
 
-def distribute_unevenly(s, recurso):
-
-	recursoTotal = 0
-	(idsCiudades, ciudades) = getIdsOfCities(s)
-	ciudadesOrigen = {}
-	ciudadesDestino = {}
-	for idCiudad in idsCiudades:
-		esTarget =  ciudades[idCiudad]['tradegood'] == str(recurso)
-		if esTarget:
-			html = s.get(urlCiudad + idCiudad)
-			ciudad = getCity(html)
-			if recurso == 1:
-				ciudad['disponible'] = ciudad['recursos'][recurso] - ciudad['consumo'] - 1
+def distribute_unevenly(session, resource_type):
+	"""
+	Parameters
+	----------
+	session : ikabot.web.session.Session
+	resource_type : int
+	"""
+	resources_total = 0
+	(cities_ids, cities) = getIdsOfCities(session)
+	origin_cities = {}
+	destination_cities = {}
+	for city_id in cities_ids:
+		is_city_mining_this_resource = cities[city_id]['tradegood'] == str(resource_type)
+		if is_city_mining_this_resource:
+			html = session.get(city_url + city_id)
+			city = getCity(html)
+			if resource_type == 1:
+				city['disponible'] = city['recursos'][resource_type] - city['consumo'] - 1
 			else:
-				ciudad['disponible'] = ciudad['recursos'][recurso]
-			if ciudad['disponible'] < 0:
-				ciudad['disponible'] = 0
-			recursoTotal += ciudad['disponible']
-			ciudadesOrigen[idCiudad] = ciudad
+				city['disponible'] = city['recursos'][resource_type]
+			if city['disponible'] < 0:
+				city['disponible'] = 0
+			resources_total += city['disponible']
+			origin_cities[city_id] = city
 		else:
-			html = s.get(urlCiudad + idCiudad)
-			ciudad = getCity(html)
-			ciudad['disponible'] = ciudad['freeSpaceForResources'][recurso]
-			if ciudad['disponible'] > 0:
-				ciudadesDestino[idCiudad] = ciudad
+			html = session.get(city_url + city_id)
+			city = getCity(html)
+			city['disponible'] = city['freeSpaceForResources'][resource_type]
+			if city['disponible'] > 0:
+				destination_cities[city_id] = city
 
-	if recursoTotal <= 0:
+	if resources_total <= 0:
 		print(_('\nThere are no resources to send.'))
 		enter()
 		return None
-	if len(ciudadesDestino) == 0:
+	if len(destination_cities) == 0:
 		print(_('\nThere is no space available to send resources.'))
 		enter()
 		return None
 
-	recursoXciudad = recursoTotal // len(ciudadesDestino)
-	espacios_disponibles = [ ciudadesDestino[city]['disponible'] for city in ciudadesDestino ]
-	totalEspacioDisponible = sum( espacios_disponibles )
-	restanteAEnviar = min(recursoTotal, totalEspacioDisponible)
+	resources_per_city = resources_total // len(destination_cities)
+	spaces_available = [ destination_cities[city]['disponible'] for city in destination_cities ]
```

Comment:
the word `spaces` looks weird there, probably should be sth related to resources and warehouse free capacity available probably
cc @physics-sp

Reply:
yes, true, maybe something like free_storage_available

Reply:
hm, is it free storage or filled storage (opposite meaning)?